cmd/changelog-parser/main.go

```diff
@@ -175,7 +150,11 @@ func collectChangelogs(repoConfig yamlRepoConfig, httpClient *http.Client) (
 				repo.Version = version
 			}
 
-			availableVersions, err := getAvailableReleases(httpClient, "github", repo.Name)
+			// eg. https://api.github.com/repos/cyberark/secretless-broker/releases
+			// This URL resoution and release fetching should be from a generic
+			// interface once we have more providers
+			releasesURL := fmt.Sprintf(providerReleasesTemplate["github"], repo.Name)
```

Comment:
I initially thought `providerReleasesTemplate` might be better suited to being inside the github package...

This was because I looked at  `github.GetAvailableReleases` and thought it odd that this was mostly a GET method that had the expertise to deal with the response but the URL is passed to it.

...but upon further inspection I realised it was one of a few maps that cater to non-github source code repositories. It hadn't dawned on me before but now I can't help but think this is a case of YAGNI (until you need it). I wonder if the value under the github key below should just live in the github package.

```
var providerToEndpointPrefix = map[string]string{
	"github": "https://raw.githubusercontent.com",
}

var providerVersionResolutionTemplate = map[string]string{
	"github": "https://api.github.com/repos/%s/releases/latest",
}

var providerReleasesTemplate = map[string]string{
	"github": "https://api.github.com/repos/%s/releases",
}
```

Reply:
@doodlesbykumbi Good point. I'll pull only the relevant map into the package to make this a less messy of a change for now but I'll keep this in mind for follow-ups.